builds/e2e/longhaul.yaml

```diff
@@ -0,0 +1,64 @@
+trigger: none
+
+jobs:
+
+################################################################################
+  - job: LinuxAMD64Moby
+################################################################################
+    displayName: Linux AMD64 Moby
+    pool:
+      name: Azure-IoT-Edge-Core
+      demands:
+        - run-long-haul -equals true
+        - Agent.OS -equals Linux
+        - Agent.OSArchitecture -equals X64
+    steps:
+      - task: DownloadBuildArtifacts@0
+        displayName: 'Download Iotedged Artifacts'
+        inputs:
+          buildType: specific
+          project: 'b32aa71e-8ed2-41b2-9d77-5bc261222004'
+          pipeline: 55463
```

Comment:
We should make the `project`, `pipeline`, and `artifactName` fields as variables.